### PR TITLE
Project improvements

### PR DESCRIPTION
**Pull request by vadimdemedes**
This PR is very similar to the one in [analytics-node](https://github.com/segmentio/analytics-node/pull/111).

#### 1. Add [editorconfig](http://editorconfig.org/)

Ensures that all contributors write code with the same formatting according to the `standard` linter. Requires `editorconfig` plugin to be installed in the user's editor, but experience shows most of them have it.

#### 2. Remove yarn.lock and disable npm package locks (`package-lock.json`)

Those are ignored by both Yarn and npm anyway, so there's no reason to include them into repository. Package locks make sense in apps, not libraries. See https://github.com/sindresorhus/ama/issues/479#issuecomment-310661514 for a more detailed explanation.

#### 3. Improve package.json

This change reorganizes fields in package.json to the "right" order and adds many useful things. Foe example, I added a `files` field (https://docs.npmjs.com/files/package.json#files) to include only `index.js` in npm package.

#### 4. Add license

Previously there was no license specified, except in package.json. I pasted license to `license` file (not `.md`, since it's not markdown, but plain text), which brings useful things to our repository, such as:

screen shot 2017-07-26 at 3 42 02 pm

and

screen shot 2017-07-26 at 3 41 44 pm

#### 5. Remove caching of dependencies on CI

It's always preferable for CI builds to be fresh, without any previous state. Sometimes this helps to catch some nasty bugs related to dependencies.

#### 6. Test on multiple Node.js versions on CI

This one is pretty explanatory, previously tests ran only for Node.js v4.x.x.

#### 7. Remove continuous deployment

After talking with @stevenmiller888, we agreed that [np](https://github.com/sindresorhus/np) should be used for publishing new releases.

#### 8. Use npm scripts instead of Makefile

Makefiles are great, but in this case the whole Makefile can be replaced with a one line in package.json.

#### 9. Remove unused dependencies

Bluebird wasn't used but was still specified as a dependency. Bye! Same for istanbul.

#### WIP - 10. Improve readme

Just prettify all the things.

### CODE REVIEW

File: license [hunk 1]
```diff
@@ -0,0 +1,7 @@
+ISC License
```
Comment (stevenmiller888) [inline license]: Should be the MIT license

Comment (vadimdemedes) [inline license]: I thought so too, but previously it was "ISC" in package.json, so maybe there was a reason for it.

Comment (vadimdemedes) [inline license]: Changed.

**Comment by vadimdemedes** [comment]
Currently this PR is blocked by https://github.com/segmentio/jsonrpc2.js/pull/18, which also modifies the readme. I want to change the format little bit of API docs, so it would collide.

File: package.json [hunk 2]
```diff
@@ -1,12 +1,24 @@
 {
   "name": "@segment/jsonrpc2",
   "version": "0.6.0",
-  "license": "ISC",
+  "description": "JSON-RPC over HTTP/TCP client",
+  "license": "MIT",
+  "repository": "segmentio/jsonrpc2",
+  "author": {
+    "name": "Segment",
+    "email": "tools+npm@segment.com",
```
Comment (stephenmathieson) [inline package.json]: probably want friends@ for this

Comment (vadimdemedes) [inline package.json]: Done.

**Comment by vadimdemedes** [comment]
I updated the readme to have a new format for API docs. I had to force push, so that the commit history is clean and we can later revert any of those changes without having to revert the whole PR.

**Pull request merged**